src/comments.js

```diff
@@ -332,7 +341,15 @@ function printTrailingComment(commentPath, print, options, parentNode) {
     // trailing comment for `2`. We can simulate the above by checking
     // if this a comment on its own line; normal trailing comments are
     // always at the end of another expression.
-    return concat([hardline, contents]);
+    if (isException) {
```

Comment:
Same concern here, the `print{Leading,Trailing}Comment` functions are written very carefully and I don't want to make them any more complicated. Let me think about this a bit.